db/query.go

```diff
@@ -500,6 +553,9 @@ func (context *DatabaseContext) QueryPrincipals(ctx context.Context, startKey st
 	}
 
 	queryStatement := replaceIndexTokensQuery(QueryPrincipals.statement, sgIndexes[IndexSyncDocs], context.UseXattrs())
+	if context.Options.Serverless {
```

Comment:
Better to use if/else here to avoid paying for token replacement for a statement we're not going to use.

Reply:
addressed